src/Avalonia.Controls/Shapes/Arc.cs

```diff
@@ -0,0 +1,172 @@
+using System;
+using Avalonia.Media;
+
+namespace Avalonia.Controls.Shapes
+{
+    public class Arc : Shape
+    {
+        ArcSegment _arcSegment;
+        PathFigure _arcFigure;
+        PathGeometry _arcGeometry;
+
+        PathGeometry _emptyGeometry = new PathGeometry()
```

Comment:
I wonder - can this be static and lazily created when needed? I can imagine an app having tons of `Arc` controls and there is no need to make arc creation so expensive.

Reply:
Why do we even need it? Can't we just return null in case there is nothing to render?

Reply:
@MarchingCube I was specifically told to *not* create `Geometry` instances more frequently than needed...I feel like I must've misunderstood what you're getting at.

Also, is there a better geometry type to use for this than `PathGeometry`? I'll switch over if so, if you can tell me what it is.

@Gillibald Seems not, as the base `Shape` class acts up with regards to layouting if `CreateDefiningGeometry` returns `null`...unless I misinterpreted what I'd seen happen? Note to self: look into that.

Reply:
Yeah, the thing about using `PathGeometry` is that it will create geometry all the time anyway. Since it basically listens to changes to segments and rebuilds inner stream geometry. And that listening part has overhead since it has to subscribe to several layers of properties.

One might just create a new `StreamGeometry` and draw to it directly. Avoiding all indirection.

About returning `null` - yeah that should be technically supported. That function returns `Geometry?` so it seems that `null` is an expected outcome.